presto-pinot/src/main/java/io/prestosql/pinot/PinotMetadata.java

```diff
@@ -267,6 +278,73 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         return Optional.of(new ConstraintApplicationResult<>(handle, constraint.getSummary()));
     }
 
+    @Override
+    public Optional<AggregationApplicationResult<ConnectorTableHandle>> applyAggregation(
+            ConnectorSession session,
+            ConnectorTableHandle handle,
+            List<AggregateFunction> aggregates,
+            Map<String, ColumnHandle> assignments,
+            List<List<ColumnHandle>> groupingSets)
+    {
+        if (!isAggregationPushdownEnabled(session)) {
+            return Optional.empty();
+        }
+
+        PinotTableHandle tableHandle = (PinotTableHandle) handle;
+
+        if (tableHandle.getQuery().isPresent()) {
+            return Optional.empty();
+        }
+        if (tableHandle.getLimit().isPresent()) {
+            // handle's limit is applied after aggregations, so we cannot apply aggregations if limit is already set
+            return Optional.empty();
+        }
+
+        // Global aggregation is represented by [[]]
+        verify(!groupingSets.isEmpty(), "No grouping sets provided");
+
+        if (groupingSets.size() != 1) {
+            return Optional.empty();
+        }
+
+        ImmutableList.Builder<ConnectorExpression> projections = ImmutableList.builder();
+        ImmutableList.Builder<Assignment> resultAssignments = ImmutableList.builder();
+        ImmutableList.Builder<AggregationExpression> aggregationExpressions = ImmutableList.builder();
+
+        for (AggregateFunction aggregate : aggregates) {
+            String inputExpression = getInputExpresion(aggregate);
+            Optional<String> aggregateFunctionName = getAggregationFunctionName(aggregate);
+            if (!aggregateFunctionName.isPresent()) {
+                return Optional.empty();
+            }
+            String outputColumnName = format("%s(%s)", aggregateFunctionName.get(), inputExpression);
```

Comment:
What role does this column name have? Will it be an issue if it contains characters like parentheses?

Reply:
It's used to build the aggregate expression, pinot names the output column as "functionName(columnName)". Currently inputExpression is a single column name, even if it contains parens it would not be an issue.

Reply:
Added tests to ensure we capture all the behavior from pinot. Found some surprising things like pinot lower cases mixed case columns when there is an aggregate function used, and there is no min(string_type) function.